File: fingertip/src/main/java/com/praszapps/easyfingerprint/presenter/FingertipDialogFragmentPresenter.java
package com.praszapps.easyfingerprint.presenter;


import com.praszapps.easyfingerprint.MVP.FingertipMVPContract;
import com.praszapps.easyfingerprint.model.repository.FingertipRepository;

import javax.inject.Inject;
import javax.inject.Named;

public class FingertipDialogFragmentPresenter implements FingertipMVPContract.Presenter {

    private FingertipMVPContract.View mView = null;
    private FingertipRepository mFingerprintRepository = null;

    @Inject
    public FingertipDialogFragmentPresenter(@Named("dialogFragmentView") FingertipMVPContract.View mView, FingertipRepository mFingerprintRepository) {
        this.mView = mView;
        this.mFingerprintRepository = mFingerprintRepository;
    }

    private FingertipDialogFragmentPresenter() {
    }

    public void init(FingertipMVPContract.View view) {
        this.mView = view;

        if(!mFingerprintRepository.isFingerprintAvailable()) {
            mView.setUpFingerprintViews();
        } else {
            mView.handleNoFingerprintAuthPossible();
        }
    }

    @Override
    public void authenticateViaFingerprint() {
        mFingerprintRepository.startFingerprintTracking();
    }

    @Override
    public void stopFingerprintTracking() {
        mFingerprintRepository.stopFingerprintTracking();
    }
}

File: janus-authenticator/src/main/java/com/praszapps/fingertip/presenter/FingertipDialogFragmentPresenter.kt
package com.praszapps.fingertip.presenter


import android.app.KeyguardManager
import android.support.v4.hardware.fingerprint.FingerprintManagerCompat
import com.praszapps.fingertip.contract.FingertipMVPContract
import com.praszapps.fingertip.model.repository.FingertipSecureModel
import kotlinx.coroutines.experimental.GlobalScope
import kotlinx.coroutines.experimental.launch

internal class FingertipDialogFragmentPresenter(IView: FingertipMVPContract.IView) : FingertipMVPContract.IPresenter {

    private val mView = IView

    private val mFingerprintRepository = FingertipSecureModel()

    private val listener = object : FingerprintManagerCompat.AuthenticationCallback() {
        override fun onAuthenticationSucceeded(result: FingerprintManagerCompat.AuthenticationResult?) {
            mView.onFingerPrintAuthenticationSuccess()
        }

        override fun onAuthenticationError(errMsgId: Int, errString: CharSequence?) {
            mView.onFingerprintAuthenticationFailed(errString.toString())
        }

        override fun onAuthenticationHelp(helpMsgId: Int, helpString: CharSequence?) {
            mView.onFingerprintAuthenticationFailed(helpString.toString())
        }

        override fun onAuthenticationFailed() {
            mView.onFingerprintAuthenticationFailed("Fingerprint not recognized. Try again")
        }
    }

    override fun initialize(mFingerprintManager: FingerprintManagerCompat, mKeyguardManager: KeyguardManager) {

        GlobalScope.launch {
            val model = mFingerprintRepository.initialize(mFingerprintManager, mKeyguardManager)

            if (model.isSuccess) {
                mView.setUpFingerprintViews()
            } else {
                mView.onFingerprintAuthenticationFailed(model.message)
            }
        }
    }

    override fun authenticateViaFingerprint() {

        mFingerprintRepository.startFingerprintTracking(listener)
    }

    override fun cancelFingerprintDetection() {
        mFingerprintRepository.stopFingerprintTracking()
    }
}